### PR TITLE
heartbeat for swift

### PR DESCRIPTION
**Pull request by jayahariv**
* implementation and unit test to validate the exception

### CODE REVIEW

**Comment by jayahariv** [comment]
PR Validation seems to be failing due to simulator/Xcode issue, meanwhile you can review this.

File: Swift/Tests/CBLTestHelper.m [hunk 1]
```diff
@@ -27,4 +27,18 @@ + (void) allowExceptionIn: (void (^)(void))block {
     --gC4ExpectExceptions;
 }
 
++ (BOOL) catchException: (void(^)(void))tryBlock error: (__autoreleasing NSError **)error {
```
Comment (pasin) [inline Swift/Tests/CBLTestHelper.m:30]: I have never specified __autoreleasing. What doesn't it do specifically for this case?

File: Swift/Tests/CBLTestHelper.m [hunk 2]
```diff
@@ -27,4 +27,18 @@ + (void) allowExceptionIn: (void (^)(void))block {
...
+    @try {
+        ++gC4ExpectExceptions;
+        tryBlock();
+        --gC4ExpectExceptions;
+        return YES;
+    }
+    @catch (NSException *exception) {
+        *error = [[NSError alloc] initWithDomain: exception.name code: 0
```
Comment (pasin) [inline Swift/Tests/CBLTestHelper.m:38]: Can it be *error = exception?

**Review by pasin — APPROVED** [review]
I have two minor comments.

File: Swift/Tests/CBLTestHelper.m (hunk 1 above)
Comment (jayahariv) [reply Swift/Tests/CBLTestHelper.m:30]: `NSError** error` implicitly qualified as `__autoreleasing `. For better clarity, the example I copied, have explicitly specified the `__autoreleasing`. I will remove it. 

https://stackoverflow.com/questions/13587742/nserror-and-autoreleasing/13590696#13590696
https://clang.llvm.org/docs/AutomaticReferenceCounting.html#indirect-parameters

File: Swift/Tests/CBLTestHelper.m (hunk 2 above)
Comment (jayahariv) [reply Swift/Tests/CBLTestHelper.m:38]: Do you mean, `*error = exception`?

It's not possible, NSError and NSException type mismatch?

**Pull request merged**